test/manipulation.test.js

```diff
@@ -193,6 +297,29 @@ describe('manipulation', function () {
       });
     });
 
+    it('should create batch of objects with beforeCreate (promise variant)', function(done) {
+      Person.beforeCreate = function(next, data) {
+        if (data && data.name === 'A') {
+          return next(null, {id: 'a', name: 'A'});
+        } else {
+          return next();
+        }
+      };
+      var batch = [
+        {name: 'A'},
+        {name: 'B'},
+        undefined
+      ];
+      Person.create(batch)
+      .then(function(ps) {
+        should.exist(ps);
+        ps.should.be.instanceOf(Array);
+        ps.should.have.lengthOf(batch.length);
+        ps[0].should.be.eql({id: 'a', name: 'A'});
+        done();
+      }).catch(done);
+    });
```

Comment:
Originally I just wanted to make sure that it worked without inspecting all
the test cases fully.
I think I just duplicated every test that I could rewrite using promises
rather than callbacks... we can probably take out a bunch of them.

I was a bit more selective in the relations tests, once I realized that
there were hundreds of them....

On Thu, Feb 19, 2015 at 10:17 AM, Miroslav Bajtoš <notifications@github.com>
wrote:

> In test/manipulation.test.js
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451#discussion_r25005722>
> :
>
> > +        }
> > +      };
> > +      var batch = [
> > +        {name: 'A'},
> > +        {name: 'B'},
> > +        undefined
> > +      ];
> > +      Person.create(batch)
> > +      .then(function(ps) {
> > +        should.exist(ps);
> > +        ps.should.be.instanceOf(Array);
> > +        ps.should.have.lengthOf(batch.length);
> > +        ps[0].should.be.eql({id: 'a', name: 'A'});
> > +        done();
> > +      }).catch(done);
> > +    });
>
> The beforeCreate hook is not really specific to the callback vs. promise
> implementation, right? In which case I prefer to omit this promise-variant
> test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451/files#r25005722>
> .
>